### ISSUE TITLE
[java] AssignmentInOperand false positives in 7.17.0

### ISSUE THREAD

**Issue by vzorge** [pmd/pmd#6075]
**Affects PMD Version:** 
7.17.0
**Rule:**
AssignmentInOperand
Please provide the rule name and a link to the rule documentation:
[AssignmentInOperand](https://docs.pmd-code.org/pmd-doc-7.17.0/pmd_rules_java_errorprone.html#assignmentinoperand)

**Description:**
The fix for issue https://github.com/pmd/pmd/issues/3434 causes version 7.17.0 to flag assignment in Consumers as part of the AssignmentInOperand. This does not seem to be the intention, as it was not mentioned in any of the tests in this fix.

We use Consumers in this way in a Reactor Mono to assign value in a `doOnNext`.

**Code Sample demonstrating the issue:**

```java
class PmdTest {
    private String field;
    public void pmdTest() {
        Consumer<String> cons = s -> this.field = s;
        cons.accept("pmd");
    }
}
```

**Expected outcome:**

PMD reports a violation at line 4, but that's wrong. That's a false positive.

**Running PMD through:** *[Gradle]*
